docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,128 @@
+# CSS variables
+
+<p class="description">Learn about the experimental API for using CSS variables on Material UI components.</p>
+
+CSS variables is the most upvoted `@mui/system` related issue.
+It allows several improvements in developer experience, like fixing dark mode SSR flickering, allowing multiple themes other than only light and dark, better debugging experience overall, and many others.
+To learn more, check [the RFC](https://github.com/mui/material-ui/issues/27651) we did about it.
+
+A while ago, we added an experimental API to the System package meant to allow styling the components with CSS variables.
+We want now to take it one step further and utilize it in one of the Material UI components.
+
+## Usage
+
+At this moment, the API does not require any breaking changes for the Material UI components, however it depends on using a new experimental provider for the theme, called `Experimental_CssVarsProvider`.
+Except for providing the theme in the inner React context, this new provider has as a responsibility to generate CSS variables out of all tokens in the theme that are not functions, and make them available in the context.
+
+All these variables, are available under a key in the theme, named `vars`.
+The structure of this object is identical to the theme structure, the only difference is that the values represent some CSS varaibles.
+
+The best way to see this is by example.
+Let's say we have this element that uses some theme token:
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivThemeToken.js", "hideToolbar": true }}
+
+If you try to inspect it, you will see that the color style has a direct hex value, that comes from some theme token:
+
+```html
+<div style="color: #2e7d32;">...</div>
+```
+
```

Comment:
I would elaborate on why this is a problem. This is based on my understanding after reading the RFC.

```suggestion

This makes the code difficult to debug, because the app could have multiple theme tokens, but this doesn't give you any information about which theme is responsible for this style.

```